src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/haflow/update/actions/UpdateFlowStatusAction.java

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.fsm.haflow.update.actions;
+
+import static java.lang.String.format;
+
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.FlowPathStatus;
+import org.openkilda.model.FlowStatus;
+import org.openkilda.model.HaFlow;
+import org.openkilda.model.HaSubFlow;
+import org.openkilda.model.PathId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.wfm.share.logger.FlowOperationsDashboardLogger;
+import org.openkilda.wfm.topology.flowhs.fsm.common.actions.FlowProcessingWithHistorySupportAction;
+import org.openkilda.wfm.topology.flowhs.fsm.haflow.update.HaFlowUpdateContext;
+import org.openkilda.wfm.topology.flowhs.fsm.haflow.update.HaFlowUpdateFsm;
+import org.openkilda.wfm.topology.flowhs.fsm.haflow.update.HaFlowUpdateFsm.Event;
+import org.openkilda.wfm.topology.flowhs.fsm.haflow.update.HaFlowUpdateFsm.State;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+public class UpdateFlowStatusAction extends
```

Comment:
I'd use HA in all related classes if they are not usable for all flows. Although they are in separate packages they will clutter the search and it's easier. Also you use HA in some classes, I suggest to unify the naming. (this is not blocking for this PR, could be done in another PR)

Reply:
You are right. I just forgot to rename classes. will do in next PR